### PR TITLE
Add ValidatingEntityType

### PR DESCRIPTION
**Pull request by cristinamanum**
Add ValidatingEntityType to the Package Validation Set and add it to the ValidationMessage Data.
This will allow:
- The ValidationMessage Handlers can choose to reject any validation types that they do not expect.
- The Package Validation Sets can be filtered only for the validation sets specific for a type.

### CODE REVIEW

File: src/NuGet.Services.Validation/PackageValidationMessageData.cs [hunk 1]
```diff
@@ -8,18 +8,28 @@ namespace NuGet.Services.Validation
 {
     public class PackageValidationMessageData
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation/PackageValidationMessageData.cs]: >PackageValidationMessageData [](start = 17, length = 28)

Should we rename this to `ValidationMessageData`?

File: src/NuGet.Services.Validation/PackageValidationMessageData.cs [hunk 2]
```diff
@@ -8,18 +8,28 @@ namespace NuGet.Services.Validation
...
     {
+        public PackageValidationMessageData(
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation/PackageValidationMessageData.cs]: >PackageValidationMessageData [](start = 15, length = 28)

Should we make the `validatingEntityType` parameter required?

File: src/NuGet.Services.Validation/Entities/PackageValidationSet.cs [hunk 3]
```diff
@@ -71,5 +71,11 @@ public class PackageValidationSet
         /// The package validations in the validation set.
         /// </summary>
         public virtual ICollection<PackageValidation> PackageValidations { get; set; }
+
+        /// <summary>
+        /// The entity type to be validated.
+        /// </summary>
+        public ValidatingEntityType ValidatingEntityType { get; set; }
```
Comment (loic-sharma) [inline src/NuGet.Services.Validation/Entities/PackageValidationSet.cs]: >ValidatingEntityType [](start = 36, length = 20)

What will the Orchestrator's query patterns be like? Do we need any indexes?

Also, I think a shorter property name could be fine here - what do you think of `Type`?

File: src/NuGet.Services.Validation/PackageValidationMessageData.cs (hunk 1 above)
Comment (cristinamanum) [reply src/NuGet.Services.Validation/PackageValidationMessageData.cs]: I think that if will be more cases when this needs to be used for other types is worth the work of renaming. Symbol is very close to the package structure, If not strong objection I will keep like it is.

File: src/NuGet.Services.Validation/PackageValidationMessageData.cs (hunk 2 above)
Comment (cristinamanum) [reply src/NuGet.Services.Validation/PackageValidationMessageData.cs]: I think that if the name type nae is PacakgeValidationMessageData having the default to package makes sense.  If not strong objection I will keep like it is.

File: src/NuGet.Services.Validation/Entities/PackageValidationSet.cs (hunk 3 above)
Comment (cristinamanum) [reply src/NuGet.Services.Validation/Entities/PackageValidationSet.cs]: Good point for the indexes.I will look into this. I shortened the name to ValidatingType .

**Pull request merged**